bricks/ui-test/src/data-providers/clone-nodes.ts

```diff
@@ -0,0 +1,61 @@
+import { createProviderClass } from "@next-core/utils/general";
+import { NodeGraphData, TreeNodeItemData, TestTreeData } from "../interface.js";
+import { findNodeData } from "./sort-test-tree-node.js";
+import { InstanceApi_createInstance } from "@next-api-sdk/cmdb-sdk";
+import { isEmpty, pick } from "lodash";
+
+const basicFields = ["name", "label", "sort", "type", "params"];
+
+async function createChildrenNode(
+  parentData: TreeNodeItemData,
+  nodes: NodeGraphData[],
+  treeData: TestTreeData[]
+): Promise<void> {
+  const responseNodes = (await Promise.all(
+    nodes.map((node) =>
+      InstanceApi_createInstance("UI_TEST_NODE@EASYOPS", {
+        ...pick(node, basicFields),
+        parent: parentData?.instanceId,
+      })
+    )
+  )) as TreeNodeItemData[];
+
+  let index = 0;
+  for (const item of nodes) {
+    const findNode = findNodeData(treeData, item.instanceId)?.data;
+
+    if (!isEmpty(findNode?.children)) {
+      await createChildrenNode(
+        responseNodes[index],
+        findNode!.children as NodeGraphData[],
+        treeData
+      );
+    }
+    index += 1;
+  }
+}
+
+export async function cloneNodes(
+  clonedData: TreeNodeItemData,
+  parentData: TreeNodeItemData,
+  treeData: TestTreeData[]
+): Promise<void> {
+  const instanceData = await InstanceApi_createInstance(
+    "UI_TEST_NODE@EASYOPS",
+    {
+      ...pick(clonedData, basicFields),
+      sort: parentData.children?.length ?? 0,
```

Comment:
这里应该要先找到最后的那个 child 的 sort，给它 +1：

```suggestion
      sort: parentData.children?.length ? (parent.children[parent.children.length - 1].sort ?? 0) + 1 : 0
```

Reply:
因为我们在删除节点的时候，不会去动其兄弟节点的 sort，比如：

- 0
- 1
- 2

我们删除中间的 `1` 后，`children.length` 为 `2`，但是下一个子节点的序号应该为 `2 + 1 = 3`